### PR TITLE
initial version of TypeScript support in jspm

### PR DESCRIPTION
**Pull request by vladima**

### CODE REVIEW

**Comment by guybedford** [comment]
:+1:

File: lib/core.js [hunk 1]
```diff
@@ -183,9 +186,17 @@ exports.dlTranspiler = function(transpilerName, update) {
...
+    
     // just do a quick install which checks basic existence
-    return install.install(installObj, { quick: !update, dev: true, summary: false });
+    var options = { quick: !update, dev: true, summary: false };
+    if (transpilerName === 'typescript') {
+      // specify shim configuration for typescript package
+      // having shim implies that 'typescript' module will have 'global' format 
+      options.override = {
+        shim: { 'bin/typescript': { 'exports': 'ts' } }
+      };
```
Comment (guybedford) [inline lib/core.js]: This needs to be committed to the jspm registry rather as a shared override - https://github.com/jspm/registry/.

Comment (vladima) [inline lib/core.js]: sure, I'll send a separate PR to the jspm registry repo

**Comment by vladima** [comment]
PR for adding TypeScript to jspm registry: https://github.com/jspm/registry/pull/377

**Comment by gdi2290** [comment]
:+1:

**Comment by guybedford** [comment]
@vladima finally merged! Thanks again for your efforts and patience. Release will follow as soon as I can this week, just IE testing and documentation to go. Happy to update the TypeScript reference when the time comes, please do update here when that happens.

**Comment by crissdev** [comment]
Awesome! I've been waiting for this. :+1:  
I'll be using the master branch until release. So far it works well.

**Comment by vladima** [comment]
@guybedford thanks! looking forward for the official release

**Comment by atrauzzi** [comment]
When the official release of TS support in JSPM happens, would it it be possible to get some of the documentation updated to discuss the workflow.  Even if it's just to outline similar things with a TS-specific bent.

@guybedford, perhaps you can shed some light on this - but right now when working with **other** TypeScript packages, one would be leveraging that package's built-output.  Not the raw TS code that might be included.

Are there any plans or is there perhaps already a convention to use the actual `.ts` or `.r.ts` files from my JSPM project's `.ts` sources?  It would be neat to be able to require an NPM package and start smashing away with strongly typed code, instead of having to assume it's all been flattened to ES5 already.

**Comment by guybedford** [comment]
@atrauzzi @frankwallis is the best person to chat to about this - I have very limited knowledge on the TypeScript workflows.

**Comment by Rush** [comment]
What's the performance of the TypeScript integration in regards to using it in a medium to large project? For example I am concerned that compiling 100 js files in the browser may be too heavy on every page open in development. I am going for max 3 seconds from open to load of my application in development.

**Pull request merged**